File: NotesServer/Models/AuthModels.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace AuthServer.Models;

public class RegisterRequest
{
    [Required]
    [EmailAddress]
    public string Email { get; set; } = string.Empty;
    
    [Required]
    [MinLength(6)]
    public string Password { get; set; } = string.Empty;
    
    [Required]
    public string DisplayName { get; set; } = string.Empty;
}

public class LoginRequest
{
    [Required]
    [EmailAddress]
    public string Email { get; set; } = string.Empty;
    
    [Required]
    public string Password { get; set; } = string.Empty;
}

public class UpdateProfileRequest
{
    public string? DisplayName { get; set; }
    [JsonPropertyName("photoURL")]
    public string? PhotoUrl { get; set; }
}

public class AuthResponse
{
    public string Token { get; set; } = string.Empty;
    public UserDto User { get; set; } = new UserDto();
}

public class UserDto
{
    public int Id { get; set; }
    public string Email { get; set; } = string.Empty;
    public string? DisplayName { get; set; }
    [JsonPropertyName("photoURL")]
    public string? PhotoUrl { get; set; }
    
    public static UserDto FromUser(User user)
    {
        return new UserDto
        {
            Id = user.Id,
            Email = user.Email,
            DisplayName = user.DisplayName,
            PhotoUrl = user.PhotoUrl
        };
    }
} 

File: NotesServerGO/models/auth_models.go
package models

// RegisterRequest представляет данные для регистрации нового пользователя.
type RegisterRequest struct {
	Email       string `json:"email"`
	Password    string `json:"password"`
	DisplayName string `json:"displayName"`
}

// LoginRequest представляет данные для входа пользователя.
type LoginRequest struct {
	Email    string `json:"email"`
	Password string `json:"password"`
}

// UserPublicInfo представляет публичные данные пользователя, возвращаемые API.
type UserPublicInfo struct {
	ID          int64  `json:"id"`
	Email       string `json:"email"`
	DisplayName string `json:"displayName"`
	PhotoUrl    string `json:"photoUrl"`
}

// AuthResponse представляет ответ сервера после успешной аутентификации.
type AuthResponse struct {
	Token string         `json:"token"`
	User  UserPublicInfo `json:"user"`
}

// UpdateProfileRequest представляет данные для обновления профиля пользователя.
type UpdateProfileRequest struct {
	DisplayName string `json:"displayName"`
	PhotoUrl    string `json:"photoUrl"`
}
